Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -508,7 +508,8 @@ protected void MuzzleFlash(EntityUid gun, AmmoComponent component, Angle worldAn
             return;
 
         var ev = new MuzzleFlashEvent(GetNetEntity(gun), sprite, worldAngle);
-        CreateEffect(gun, ev, gun);
+        var sending = _netManager.IsServer && user != null ? user : gun;
+        CreateEffect(gun, ev, sending);
```

Comment:
Pretty sure this could and probably should just be
```suggestion
        CreateEffect(gun, ev, user ?? gun);
```

Reply:
Apparently they're meant to be different methods on the client & server? In which case that really needs to be fixed, they shouldn't be overriding the same base method if they perform different functions

Reply:
Yeah nevermind this isn't right